test/OpenTelemetry.Tests/Trace/BatchExportActivityProcessorTest.cs

```diff
@@ -0,0 +1,142 @@
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), maxExportBatchSize: 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), maxExportBatchSize: 2049));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), scheduledDelayMilliseconds: 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), exporterTimeoutMilliseconds: -1));
+        }
+
+        [Fact]
+        public void CheckIfBatchIsExportingOnQueueLimit()
+        {
+            using var exporter = new TestActivityExporter();
+            using var processor = new BatchExportActivityProcessor(exporter, maxQueueSize: 1, maxExportBatchSize: 1);
+            processor.OnStart(new Activity("start"));
+            processor.OnEnd(new Activity("start"));
+
+            int i = 0;
+            while (i < 10)
+            {
+                if (exporter.Exported.Count == 0)
+                {
+                    Thread.Sleep(500);
+                    i++;
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            Assert.Single(exporter.Exported);
+
+            Assert.Equal(1, processor.ProcessedCount);
+            Assert.Equal(1, processor.ReceivedCount);
+            Assert.Equal(0, processor.DroppedCount);
+        }
+
+        [Fact]
+        public void CheckForceFlushWithInvalidTimeout()
+        {
+            using var exporter = new TestActivityExporter();
+            using var processor = new BatchExportActivityProcessor(exporter, maxQueueSize: 2, maxExportBatchSize: 1);
+            Assert.Throws<ArgumentOutOfRangeException>(() => processor.ForceFlush(-2));
+        }
+
+        [Theory]
+        [InlineData(Timeout.Infinite)]
+        [InlineData(0)]
+        [InlineData(1)]
+        public void CheckForceFlushExport(int timeout)
+        {
+            using var exporter = new TestActivityExporter();
+            using var processor = new BatchExportActivityProcessor(
+                exporter,
+                maxQueueSize: 3,
+                maxExportBatchSize: 3,
+                exporterTimeoutMilliseconds: 30000);
+
+            processor.OnStart(new Activity("start1"));
+            processor.OnEnd(new Activity("start1"));
+
+            processor.OnStart(new Activity("start2"));
+            processor.OnEnd(new Activity("start2"));
+
+            Assert.Equal(0, processor.ProcessedCount);
+
+            // waiting to see if time is triggering the exporter
+            Thread.Sleep(1_000);
+            Assert.Empty(exporter.Exported);
+
+            // forcing flush
+            processor.ForceFlush(timeout);
+
+            Thread.Sleep(1_000);
```

Comment:
I think we want to surround this by an if statement?

```csharp
if (timeout == 0)
{
    Thread.Sleep(1000); // ForceFlush(0) will trigger flush and return immediately, so let's sleep for a while
}
```

Reply:
updating! thanks for reviewing!